gradle/prCheck.gradle

```diff
@@ -63,48 +61,55 @@ def checkPull(changeId) {
+        return ["The associated GitHub issue #${issueId} must be mentioned in the description of the PR."]
+    } 
+        
+    def issueDetails = jsonSlurper.parseText(new URL("https://api.github.com/repos/h2oai/h2o-3/issues/${issueId}").getText())
+    
+    if (issueDetails.milestone == null || issueDetails.milestone.state != "open"){
+        return ["The associated GitHub issue #${issueId} must be assigned to an opened milestone."]
     }
-    if (fixVersions.size() > 1) {
-        return ["Pull Request ${changeId} is linked to jira ticket https://h2oai.atlassian.net/browse/${jiraId}. " +
-                "This jira has more than one fix version. Please check if the associated jira ticket is correct and validate with the team what should be the correct fixed version."]
+    if (issueDetails.assignee == null){
+        return ["The associated GitHub issue #${issueId} must  have an assignee."]
     }
-    def fixVersionValue = fixVersions.get(0).name
+    
+    def fixVersionValue = issueDetails.milestone.title
     def baseBranch = pullDetails.base.ref
     if (fixVersionValue.endsWith('.1')) {
         if (baseBranch != 'master')
-            return ["Pull Request ${changeId} is linked to jira ticket https://h2oai.atlassian.net/browse/${jiraId}. " +
-                    "This jira is marked for release in {$fixVersionValue}. This release will be done from a master branch. Change the base branch of the PR to 'master'."]
+            return ["Pull Request ${changeId} is linked to the GitHub issue https://github.com/h2oai/h2o-3/issues/${issueId}. " +
+                    "This issue is marked for release in {$fixVersionValue}. This release will be done from a master branch. Change the base branch of the PR to 'master'."]
     } else {
         if (!baseBranch.startsWith('rel-')) {
-            return ["Pull Request ${changeId} is linked to jira ticket https://h2oai.atlassian.net/browse/${jiraId}. " +
-                    "This jira is marked for release in {$fixVersionValue}. This release will be done from a fix release branch. Change the base branch to a correspond to the right fix release branch."]
+            return ["Pull Request ${changeId} is linked to the GitHub issue https://github.com/h2oai/h2o-3/issues/${issueId}. " +
+                    "This issue is marked for release in {$fixVersionValue}. This release will be done from a fix release branch. Change the base branch to a correspond to the right fix release branch."]
         }
     }
-    def productionImpact = jiraDetails.fields.customfield_12678
-    if (!productionImpact) {
-        return ["Jira ticket https://h2oai.atlassian.net/browse/${jiraId} doesn't specify impact of this change on production model use. Please fill in the field 'Production Impact'."]
+    
+    def projectRequest = '{"query":"{ repository(owner: \\"h2oai\\", name: \\"h2o-3\\") {issue(number: ' + "${issueId}" + ') {projectV2(number: 112){number}}}}"}'
```

Comment:
I believe this can be done together with milestone existance and assignee.

We can report all the issues at once.